raft/log.go

```diff
@@ -195,13 +199,18 @@ func (l *raftLog) nextCommittedEnts() (ents []pb.Entry) {
 // hasNextCommittedEnts returns if there is any available entries for execution.
 // This is a fast check without heavy raftLog.slice() in nextCommittedEnts().
 func (l *raftLog) hasNextCommittedEnts() bool {
-	off := max(l.applied+1, l.firstIndex())
-	return l.committed+1 > off
+	if l.hasPendingSnapshot() {
+		// If we have a snapshot to apply, don't also return any committed
+		// entries. Doing so raises questions about what should be applied
+		// first.
+		return false
+	}
+	return l.committed > l.applied
```

Comment:
It seems that `l.applied` will never smaller than `l.firstIndex()`, so the change makes sense to me. But I am curious why previous `max(l.applied+1, l.firstIndex()) `was added.

Reply:
I think the first index only had an effect when there was an unstable snapshot, as this could bump the first index above the applied index. I was happy to see the simplification that avoids this special case.

Reply:
Thanks for the clarification, which makes sense to me.